cmd/argoexec/commands/wait.go

```diff
@@ -35,7 +36,10 @@ func waitContainer(ctx context.Context) error {
 	if err != nil {
 		wfExecutor.AddError(err)
 	}
-	ctx = context.Background() // don't allow cancellation to impact capture of results, parameters,or artifacts
+
+	// Don't allow cancellation to impact capture of results, parameters, or artifacts
+	ctx = context.Background()
+
```

Comment:
Regarding your comment related to context: good finding. I think the call to `wfExecutor.Wait(ctx)` above still needs `ctx` to be set as it was. So, is the solution to maybe pass `context.Background()` into your defer as an argument? (which makes me wonder if the other defer there should as well)

Reply:
Yeah, that sounds right.

Reply:
Are you ok with the following? Note the use of bgCtx

```
func waitContainer(ctx context.Context) error {
	wfExecutor := initExecutor()
	
	// Don't allow cancellation to impact capture of results, parameters, artifacts, or defers.
	bgCtx := context.Background()

	defer wfExecutor.HandleError(bgCtx)    // Must be placed at the bottom of defers stack.
	defer wfExecutor.FinalizeOutput(bgCtx) // Ensures the LabelKeyReportOutputsCompleted is set to true.
	defer stats.LogStats()
	stats.StartStatsTicker(5 * time.Minute)

	// Wait for main container to complete
	err := wfExecutor.Wait(ctx)
	if err != nil {
		wfExecutor.AddError(err)
	}

	// Capture output script result
	err = wfExecutor.CaptureScriptResult(bgCtx)
	if err != nil {
		wfExecutor.AddError(err)
	}

	// Saving output parameters
	err = wfExecutor.SaveParameters(bgCtx)
	if err != nil {
		wfExecutor.AddError(err)
	}

	// Saving output artifacts
	err = wfExecutor.SaveArtifacts(bgCtx)
	if err != nil {
		wfExecutor.AddError(err)
	}
 
	// Save log artifacts
	logArtifacts := wfExecutor.SaveLogs(bgCtx)

	// Try to upsert TaskResult. If it fails, we will try to update the Pod's Annotations
	wfExecutor.ReportOutputs(bgCtx, logArtifacts)

	return wfExecutor.HasError()
}
```